frontend/src/pages/SelectCar.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import "./style/SelectCar.scss";
+
+const { apiSite } = require("../conf");
+
+export default function SelectCar(props) {
+  const [immatriculation, setImmatriculation] = useState("");
+  const [année, setAnnée] = useState("");
+  const [vin, setVin] = useState("");
+  const [kilometrage, setKilometrage] = useState("");
+  const [modele, setModele] = useState({});
+  const [modeles, initModeles] = useState([]);
+  const year = new Date().getFullYear();
+
+  useEffect(() => {
+    axios.get(`${apiSite}/modelevehicule/marque`).then(({ data }) => {
+      initModeles(data);
+      limitStringLength();
+    });
+  }, []);
+
+  function addCar(e) {
+    e.preventDefault();
+    if (vin.length !== 17) {
+      //insert toast
+    } else if (parseInt(kilometrage) >= Math.pow(10, 6) || isNaN(kilometrage)) {
+      //insert toast
+    } else if (
+      parseInt(année) < 1900 ||
+      parseInt(année) > year ||
+      isNaN(année)
+    ) {
+      //insert toast
+    } else {
+      //insert toast
+    }
+  }
+
+  function limitStringLength() {
```

Comment:
I don't get this function, what is it supposed to do ?

Reply:
this function (" limitStringLength ") is to limit the number of entry string in the input text by their restriction value.
ex: "année" has maxLength="4" so we can't entry a number of 5 strings in the input.
"Vin" has maxLength="17"
"km" has maxLength="7"